tests/components/smhi/conftest.py

```diff
@@ -1,25 +1,131 @@
 from homeassistant.components.smhi.const import DOMAIN
+from homeassistant.const import CONF_LATITUDE, CONF_LOCATION, CONF_LONGITUDE, Platform
+from homeassistant.core import HomeAssistant
+
+from . import TEST_CONFIG
+
+from tests.common import MockConfigEntry, load_fixture
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock]:
+    """Override async_setup_entry."""
+    with patch(
+        "homeassistant.components.smhi.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        yield mock_setup_entry
+
+
+@pytest.fixture(name="load_platforms")
+async def patch_platform_constant() -> list[Platform]:
+    """Return list of platforms to load."""
+    return PLATFORMS
+
+
+@pytest.fixture
+async def load_int(
+    hass: HomeAssistant,
+    mock_client: SMHIPointForecast,
+    load_platforms: list[Platform],
+) -> MockConfigEntry:
+    """Set up the SMHI integration."""
+    hass.config.latitude = "59.32624"
+    hass.config.longitude = "17.84197"
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=TEST_CONFIG,
+        entry_id="1",
+        unique_id="59.32624-17.84197",
+        version=3,
+        title="Test",
+    )
+
+    config_entry.add_to_hass(hass)
+
+    with patch("homeassistant.components.smhi.PLATFORMS", load_platforms):
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    return config_entry
+
+
+@pytest.fixture(name="mock_client")
+async def get_client(
+    hass: HomeAssistant,
+    get_data: tuple[list[SMHIForecast], list[SMHIForecast], list[SMHIForecast]],
+) -> AsyncGenerator[MagicMock]:
+    """Mock SMHIPointForecast client."""
+
+    with patch(
+        "homeassistant.components.smhi.coordinator.SMHIPointForecast",
+        autospec=True,
+    ) as mock_client:
+        client = mock_client.return_value
+        client.async_get_daily_forecast.return_value = get_data[0]
+        client.async_get_twice_daily_forecast.return_value = get_data[1]
+        client.async_get_hourly_forecast.return_value = get_data[2]
+        yield client
+
+
+@pytest.fixture(name="get_data")
+async def get_data_from_library(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    load_json: dict[str, Any],
+) -> AsyncGenerator[tuple[list[SMHIForecast], list[SMHIForecast], list[SMHIForecast]]]:
+    """Get data from api."""
+    client = SMHIPointForecast(
+        TEST_CONFIG[CONF_LOCATION][CONF_LONGITUDE],
+        TEST_CONFIG[CONF_LOCATION][CONF_LATITUDE],
+        aioclient_mock.create_session(hass.loop),
+    )
+    with patch.object(
+        client._api,
+        "async_get_data",
+        return_value=load_json,
+    ):
+        data_daily = await client.async_get_daily_forecast()
+        data_twice_daily = await client.async_get_twice_daily_forecast()
+        data_hourly = await client.async_get_hourly_forecast()
```

Comment:
So wait, we patch something in the library to make sure it processes 3 files we load via JSON?

Reply:
No, it processes the same file (first file below).
The other files are only loaded so we can use them later in the tests when we need to switch to another json file.

Reply:
Why dont we load them when we need them? And why do we patch the library get request thingy

Reply:
Because that creates additional overhead than to just load them once and then we have them ready for tests that needs them.
Not sure what you mean by the other comment but we patch the get request to input our json instead. then the other calls which are returned are manipulating the data.
It would be much more effort to patch the returns of those instead of the data collection.